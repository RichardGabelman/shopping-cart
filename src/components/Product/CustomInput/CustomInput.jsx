import { useState } from "react";
import styles from "./CustomInput.module.css";

function CustomInput({ amount, setAmount }) {
  const updateAmount = (delta) => {
    setAmount((prevValue) => Math.max(prevValue + delta, 0));
  };

  return (
    <div className={styles.qtyContainer}>
      <button onClick={() => updateAmount(-1)}>-</button>
      <input
        className={styles.input}
        type="number"
        value={amount}
        onChange={(event) => setAmount(Math.max(event.target.value, 0))}
      />
      <button onClick={() => updateAmount(1)}>+</button>
    </div>
  );
}

export default CustomInput;
