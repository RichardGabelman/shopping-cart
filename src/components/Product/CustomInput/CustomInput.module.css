.qtyContainer {
  display: flex;
}

.input {
  width: 5ch;
  text-align: center;
}

.input::-webkit-inner-spin-button, 
.input::-webkit-outer-spin-button { 
  -webkit-appearance: none; 
  margin: 0; 
}