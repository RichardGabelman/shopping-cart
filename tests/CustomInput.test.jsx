import { describe, it, expect, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import CustomInput from "../src/components/Product/CustomInput/CustomInput";

describe("CustomInput", () => {
  it("renders correctly with the initial amount", () => {
    const setAmount = vi.fn();
    const initialAmount = 5;

    render(<CustomInput amount={initialAmount} setAmount={setAmount} />);

    const input = screen.getByRole("spinbutton");
    expect(input).toHaveValue(initialAmount);
  });
});