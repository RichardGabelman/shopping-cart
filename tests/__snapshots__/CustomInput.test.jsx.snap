// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CustomInput > renders correctly (snapshot) 1`] = `
<div>
  <div
    class="_qtyContainer_e2dd4f"
  >
    <button>
      -
    </button>
    <input
      class="_input_e2dd4f"
      type="number"
      value="0"
    />
    <button>
      +
    </button>
  </div>
</div>
`;
